### PR TITLE
fix /test command

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Fix the built-in `/test` command to "work" -- it seems to be a relic from well before Client4 and was never migrated as such. The bulk of the changes involve propagating errors up the call stack to surface issues more clearly, plus an explicit fix for `/test setup teams fuzz 10 20 50`.

No guarantee is made that all the other subcommands work as expected, though I've tested several. Issues should be filed separately. Ideally, we'd rewrite this altogether to avoid use of Client4 and perhaps just use the plugin API.

#### Ticket Link
Fixes: https://mattermost.atlassian.net/browse/MM-21732

### CODE REVIEW

File: app/command_loadtest.go [hunk 1]
```diff
@@ -194,12 +206,16 @@ func (me *LoadTestProvider) SetupCommand(a *App, args *model.CommandArgs, messag
 		}
 	}
 	client := model.NewAPIv4Client(args.SiteURL)
+	client.SetToken(args.Session.Token)
```
Comment (lieut-data) [inline app/command_loadtest.go]: This was the primary source of the problem: no token was provided for the initial `CreateBasicUser` call on line 212 below.

File: app/auto_channels.go [hunk 2]
```diff
@@ -50,24 +50,22 @@ func (cfg *AutoChannelCreator) createRandomChannel() (*model.Channel, bool) {
...
 	channels := make([]*model.Channel, numChannels)
 
 	for i := 0; i < numChannels; i++ {
-		var err bool
+		var err error
 		channels[i], err = cfg.createRandomChannel()
-		if !err {
-			return channels, false
+		if err != nil {
+			return channels, err
```
Comment (reflog) [inline app/auto_channels.go]: for consistency I think this should be `return nil, err`

File: app/auto_teams.go [hunk 3]
```diff
@@ -57,24 +57,24 @@ func (cfg *AutoTeamCreator) createRandomTeam() (*model.Team, bool) {
...
 	teams := make([]*model.Team, numTeams)
 
 	for i := 0; i < numTeams; i++ {
-		var err bool
+		var err error
 		teams[i], err = cfg.createRandomTeam()
-		if !err {
-			return teams, false
+		if err != nil {
+			return teams, err
```
Comment (reflog) [inline app/auto_teams.go]: this should probably be `return nil, err`

File: app/auto_teams.go [hunk 4]
```diff
@@ -37,14 +37,14 @@ func NewAutoTeamCreator(client *model.Client4) *AutoTeamCreator {
...
-func (cfg *AutoTeamCreator) createRandomTeam() (*model.Team, bool) {
+func (cfg *AutoTeamCreator) createRandomTeam() (*model.Team, error) {
 	var teamEmail string
 	var teamDisplayName string
 	var teamName string
 	if cfg.Fuzzy {
 		teamEmail = "success+" + model.NewId() + "simulator.amazonses.com"
 		teamDisplayName = utils.FuzzName()
-		teamName = utils.FuzzName()
+		teamName = utils.RandomName(cfg.NameLength, cfg.NameCharset) + model.NewId()
```
Comment (reflog) [inline app/auto_teams.go]: out of curiosity, why not FuzzName?

File: app/auto_users.go [hunk 5]
```diff
@@ -76,36 +78,34 @@ func (cfg *AutoUserCreator) createRandomUser() (*model.User, bool) {
...
 	users := make([]*model.User, numUsers)
 
 	for i := 0; i < numUsers; i++ {
-		var err bool
+		var err error
 		users[i], err = cfg.createRandomUser()
-		if !err {
-			return users, false
+		if err != nil {
+			return users, err
```
Comment (reflog) [inline app/auto_users.go]: same comment here

File: app/auto_teams.go (hunk 4 above)
Comment (lieut-data) [reply app/auto_teams.go]: Ah, I'd forgotten this change -- `utils.FuzzName` generates decidedly invalid team names, both in length and number of characters. I have no idea how this ever worked.

File: app/auto_channels.go (hunk 2 above)
Comment (lieut-data) [reply app/auto_channels.go]:
```suggestion
			return nil, err
```

File: app/auto_teams.go (hunk 3 above)
Comment (lieut-data) [reply app/auto_teams.go]:
```suggestion
			return nil, err
```

File: app/auto_users.go (hunk 5 above)
Comment (lieut-data) [reply app/auto_users.go]:
```suggestion
			return nil, err
```

File: app/auto_teams.go (hunk 4 above)
Comment (cpoile) [reply app/auto_teams.go]: IIRC there's a new random team name utility fn in the server code, since we ran into this last time I was on SET.

Comment (lieut-data) [reply app/auto_teams.go]: I found this in the `model` package:
https://github.com/mattermost/mattermost-server/blob/9a531fa0fbb5e8592739c21ba1553a622e71cb73/model/utils.go#L149-L156

but it looks like it's just amounts to `model.NewId()` in the end as well. @cpoile, do you know if there was a "better" one?

Comment (cpoile) [reply app/auto_teams.go]: I think that is the better one -- it generates valid team names (the SET problem was that random team names would sometimes include reserved words).

Comment (lieut-data) [inline app/auto_teams.go]:
```suggestion
		teamName = model.NewRandomTeamName()
```

**Comment by lieut-data** [comment]
@cpoile, I've promoted you to a full reviewer since you've already been so kind to take a look at part of the code :)

**Pull request merged**